gemfire-core/src/main/java/com/gemstone/gemfire/distributed/internal/membership/jgroup/JGroupMembershipManager.java

```diff
@@ -3301,7 +3301,7 @@ public Set send(InternalDistributedMember[] destinations,
           short version = mbr.getAddress().getVersionOrdinal();
           if (!messages.contains(version)) {
             Message jmsg = createJGMessage(msg, local, version);
-            messages.put(version, jmsg);
+            messages.justPut(version, jmsg);
```

Comment:
Using justPut instead of put is a general performance guideline in kolokobe docs so changed those classes accordingly and their usages. The difference is mostly in having to read the old value versus not.